packages/@aws-cdk/aws-cognito/README.md

```diff
@@ -183,7 +183,12 @@ new UserPool(this, 'myuserpool', {
 As shown in the code snippet, there are data types that are available for custom attributes. The 'String' and 'Number'
 data types allow for further constraints on their length and values, respectively.
 
-Custom attributes cannot be marked as required.
+All custom attributes share the common properties `developerOnly` and `mutable`.
+
+ - `developerOnly` means that this attribute can only be modified by an administrator. The use of this property is discouraged
+ in favour of the use of write permissions for attributes in the user pool client (see [AppClients](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-client-apps.html)),
+
+ - `mutable` allows the value to be changed after the value is set by the user.
```

Comment:
Does setting `mutable` to `false` mean that even administrators can't modify it? 

In general, is there a combination of `mutable` and `developerOnly` that is invalid or does not make sense?

Reply:
Great question.
Yes, setting `mutable` to `false` means that even administrators can't modify it but it still make sense to keep both properties in case the users are allowed to sign up by themselves.

If only admins (==developers) can create new users than `mutable && developerOnly` has the same meaning of `mutable` because the attribute must be set on creation and only developers can do it.

If sign up is enabled, then `mutable && developerOnly` on an attribute means that only users created via the `AdminCreateUser` api can have that attribute set while users created via the `SignUp` API will never be able to set it.